### PR TITLE
Update UserGuide Memory and Undo-Redo features

### LINKED ISSUE #70: Update User Guide
- Updating the Draft User Guide done in Week 6 and put it in the repo.

### PR DESCRIPTION
**Pull request by zenatrick**

### CODE REVIEW

**Review by jwee97 — DISMISSED** [review]
Looks good

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+* `INDEX` refers to the index number shown in the displayed diary entry list. `INDEX` *must be a positive integer* 1, 2, 3, ...
+****
 
-=== Import and export of data [Coming in V2.0]
+Example:
+
+* `deleteentry 10` +
+Deletes the 10th diary entry.
+
+==== Locate diary entry by keywords: `find`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:268]:
```suggestion
==== Locate diary entries by keywords: `find`
```

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+Locate existing diary entries by keywords. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:269]:
```suggestion
Locates existing diary entries by keywords. +
```

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+****
+
+Example:
+
+* `addphoto caption/Sanur Beach path/photos/SanurBeach.jpg` +
+Adds an image named `SanurBeach.jpg` in the folder named `photos` in the same directory as this application
+to the photo album with the caption: "Sanur Beach".
+
+==== Set passcode for memories: `setpasscode`
+Set a passcode for the memories tab for user to lock and unlock their data. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:298]:
```suggestion
Sets a passcode for the Memory tab for user to lock and unlock the Memory tab features. +
```

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+* `RELATIVE_PATH_OF_PHOTO` only accepts a `.png` or `.jpg` image file.
+****
+
+Example:
+
+* `addphoto caption/Sanur Beach path/photos/SanurBeach.jpg` +
+Adds an image named `SanurBeach.jpg` in the folder named `photos` in the same directory as this application
+to the photo album with the caption: "Sanur Beach".
+
+==== Set passcode for memories: `setpasscode`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:297]:
```suggestion
==== Set passcode for the Memory tab: `setpasscode`
```

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+
+[WARNING]
+Passcode recovery system will only be implemented in V2.0. If the user forgets the passcode, there is no way to recovery it.
+
+Example:
+* `setpasscode l0NgP455w0Rd`
 
-Exits the program. +
-Format: `exit`
+==== Change passcode for memories: `changepasscode`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:315]:
```suggestion
==== Change passcode for the Memory tab: `changepasscode`
```

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+Change the passcode for the memories tab. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:316]:
```suggestion
Changes the passcode for the Memory tab. +
```

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+****
+* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
+* This command can only be used after setting a passcode for the `Memory` tab using the command `setpasscode`.
+* `OLD_PASSCODE` and `NEW_PASSCODE` can only be made up of at most 16 alphanumeric characters each.
+****
+
+Example:
+* `changepasscode l0NgP455w0Rd l0Ng3rP455w0Rd`
+
+==== Lock memories: `lock`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:328]:
```suggestion
==== Lock the Memory tab: `lock`
```

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
+* This command can only be used after setting a passcode for the `Memory` tab using the command `setpasscode`.
+* `lock` will disable all commands related to the `Memory` tab other than the command `unlock`.
+****
+
+Example:
+
+* `lock`
+
+==== Unlock memories: `unlock`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:342]:
```suggestion
==== Unlock the Memory tab: `unlock`
```

File: docs/UserGuide.adoc [hunk 9]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
+* This command can only be used after setting a passcode for the `Memory` tab using the command `setpasscode`.
+* `OLD_PASSCODE` and `NEW_PASSCODE` can only be made up of at most 16 alphanumeric characters each.
+****
+
+Example:
+* `changepasscode l0NgP455w0Rd l0Ng3rP455w0Rd`
+
+==== Lock memories: `lock`
+Locks the ability to view and edit the diary and photo album in the `Memory` tab. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:329]:
```suggestion
Locks the ability to view and edit the diary and photo albums in the `Memory` tab. +
```

File: docs/UserGuide.adoc [hunk 10]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+
+Example:
+
+* `unlock l0Ng3rP455w0Rd` +
+Used "l0Ng3rP455w0Rd" as the `PASSCODE` to unlock the restrictions enforced by the command `lock`.
+
+==== Export the diary to pdf file [Coming in V2.0]
+
+=== Undo-Redo
+Allows the user to revert their actions quickly and conveniently.
```
Comment (zenatrick) [inline docs/UserGuide.adoc:360]:
```suggestion
Allows the user to undo and redo their input commands quickly and conveniently.
```

File: docs/UserGuide.adoc [hunk 11]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+
+==== Undo an action: `undo`
+Undo an undoable command entered by the user. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:363]:
```suggestion
Reverts the effect of the previous input command entered by the user. +
```

File: docs/UserGuide.adoc [hunk 12]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+
+* `unlock l0Ng3rP455w0Rd` +
+Used "l0Ng3rP455w0Rd" as the `PASSCODE` to unlock the restrictions enforced by the command `lock`.
+
+==== Export the diary to pdf file [Coming in V2.0]
+
+=== Undo-Redo
+Allows the user to revert their actions quickly and conveniently.
+
+==== Undo an action: `undo`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:362]:
```suggestion
==== Undo the previous input command: `undo`
```

File: docs/UserGuide.adoc [hunk 13]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+Undo an undoable command entered by the user. +
+Format: `undo`
+
+Example:
+
+* `undo`
+
+==== Redo an action: `redo`
+Redo an undone action after using the command `undo`. +
```
Comment (zenatrick) [inline docs/UserGuide.adoc:371]:
```suggestion
Re-performs an undone input command after using the command `undo`. +
```

File: docs/UserGuide.adoc [hunk 14]
```diff
@@ -109,57 +109,176 @@ Edits the phone number and email address of the 1st person to be `91234567` and
...
+
+==== Undo an action: `undo`
+Undo an undoable command entered by the user. +
+Format: `undo`
+
+Example:
+
+* `undo`
+
+==== Redo an action: `redo`
```
Comment (zenatrick) [inline docs/UserGuide.adoc:370]:
```suggestion
==== Redo an undone input command: `redo`
```

**Review by zenatrick — COMMENTED** [review]
Grammar fixes suggestions.

**Review by joshlay96 — DISMISSED** [review]
Looks good

**Review by joshlay96 — APPROVED** [review]
Looks good!

**Pull request merged**